fedimint-server/src/net/api.rs

```diff
@@ -277,6 +279,42 @@ impl ConsensusApi {
         ))
     }
 
+    async fn get_guardian_config_backup(&self) -> ApiResult<GuardianConfigBackup> {
+        let mut tar_archive_builder = tar::Builder::new(Vec::new());
+
+        let mut append = |name: &str, data: &[u8]| {
+            let mut header = tar::Header::new_gnu();
+            header.set_path(name).expect("Error setting path");
+            header.set_size(data.len() as u64);
+            header.set_cksum();
+            tar_archive_builder
+                .append(&header, data)
+                .expect("Error adding data to tar archive");
+        };
+
+        append(
+            "local.json",
+            &serde_json::to_vec(&self.cfg.local).expect("Error encoding local config"),
+        );
+
+        append(
+            "consensus.json",
+            &serde_json::to_vec(&self.cfg.consensus).expect("Error encoding consensus config"),
+        );
+
+        // TODO: encrypt
+        append(
+            "private.json",
+            &serde_json::to_vec(&self.cfg.private).expect("Error encoding private config"),
+        );
```

Comment:
Encrypt using password given for auth. This means the salt/encrypted data of the downloaded backup will differ from the on-disk version, but the password used will still be the guardian password.